merge.py

```diff
@@ -1,98 +1,303 @@
+                    'desc': 'Add Remote',
+                    'error': 'Failed to add remote. Please try again.',
+                },
+                {
+                    'cmd': 'git remote rename origin github',
+                    'desc': 'Rename Origin',
+                    'error': 'Failed to rename origin. Please try again.',
+                },
+                {
+                    'cmd': 'git config --local --add remote.github.fetch '
+                           '"+refs/pull/*/head:refs/remotes/github/pr/*"',
+                    'desc': 'Configure git fetch.',
+                    'error': 'Failed to configure git fetch. Please try again.',
+                    'shell': True,
+                },
+                {
+                    'cmd': 'git fetch --all',
+                    'desc': 'Fetch everything.',
+                    'error': 'Fetch failed. Please try again.',
+                },
+                {
+                    'cmd': 'git checkout -b {pr_name} github/pr/{pr_num}',
+                    'desc': 'Checkout PR',
+                    'error': 'Failed to check out PR. Please try again.',
+                },
+                {
+                    'cmd': 'git rebase {branch_path}',
+                    'desc': 'Rebase against target branch.',
+                    'error': 'Automatic rebase failed. Please manually rebase '
+                             'against {branch_path} and try again.',
+                },
+                {
+                    'cmd': 'git checkout {branch_path}',
+                    'desc': 'Check out target branch.',
+                    'error': 'Failed to check out {branch_path}. Please try '
+                             'again.',
+                },
+                {
+                    'cmd': 'git merge --no-ff -m "{msg}" {pr_name}',
+                    'desc': 'Merge PR',
+                    'error': 'Merge was not successful. Please try again.',
+                },
+                {
+                    'cmd': self.config['verification_command'],
+                    'desc': 'Verifying PR',
+                    'error': 'Verification failed. Please check the error log'
+                             ' and try again.',
+                },
+                # {
+                #    'cmd': 'git push {remote_name} HEAD:{branch}',
+                #    'desc': 'Push to remote master.',
+                #    'error': 'Remote push failed. Please try again.',
+                # },
+            ]
+            for command in cmds:
+                shell = True if 'shell' in command else False
+                self.run_cmd(command['cmd'], pr_vars, command['desc'],
+                             command['error'], pr, tmp_dir, m_l, shell=shell)
+        except AssertionError as err:
+            m_l.error(err)
+            return True
+        except EnvironmentError as err:
+            m_l.error("Couldn't post comment to github. Leaving this on "
```

Comment:
What is "this" that is being left on the queue? Why not "Leaving PR {pr_num} on the queue to try again"?

Also, are we sure we can try again? What if we failed to post at the final step (pushing) but the push already went through?